### PR TITLE
Remove all fs Sync functions from async code blocks and uses async-al…

### LINKED ISSUE #34: Resolve Suggestions According to ToB Report 6, MAY 2024
### Description

3\. Missing certificate validation (Med)
4\. Synchronous function calls inside asynchronous functions (Info)
5\. Hardcoded credentials (Low)
6\. Use of outdated libraries (Info)
7\. Stack traces in Express are not disabled (Info)
8\. Docker Compose ports exposed on all interfaces (Info)

For Item 7, this can be resolved by adding flag in AWS deployment script, which is not included in this Repo

### PR DESCRIPTION
**Pull request by Phanco**
### What was the problem?

This PR resolves #34, Item 4

### How was it solved?
These functions inside `async` blocks are replaced
`fs.readFileSync` -> `fs.promises.readFile`
`fs.writeFileSync` -> `fs.promises.writeFile`
`fs.fileExists` -> `fs.promises.access` or Handled by `readFile`

### How was it tested?
- Unit tests updated
- Tree Builder and Backend re-run from the beginning locally without any problem

### CODE REVIEW

**Comment by Phanco** [comment]
P.S. according to node [documentation](https://nodejs.org/api/fs.html#fsaccesspath-mode-callback). Sync version of `fs.existsSync` would be `.access`

Also, it mentioned

> Do not use fs.access() to check for the accessibility of a file before calling fs.open(), fs.readFile(), or fs.writeFile(). Doing so introduces a race condition, since other processes may change the file's state between the two calls. Instead, user code should open/read/write the file directly and handle the error raised if the file is not accessible.

So at some areas, `fs.existsSync` is completely removed, and the check is handled properly by `readFile`

P.P.S: Not all `fs.*Sync` functions are removed, only Sync functions within `async () {}` blocks are changed

File: packages/claim-backend/test/utils/leaf-map.test.ts [hunk 1]
```diff
@@ -1,53 +1,50 @@
 import { expect } from 'chai';
 import * as sinon from 'sinon';
 import { address } from '@liskhq/lisk-cryptography';
-import * as fsHelper from '../../src/utils/fs-helper';
 import { getLeafMap, getMultisigMap, loadMerkleTree } from '../../src/utils/leaf-map';
 import { buildMockLeaf, randomHash, randomPublicKeyBuffer } from '../utils';
 import { append0x } from '../../src/utils';
 
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const fs = require('fs');
```
Comment (Phanco) [inline packages/claim-backend/test/utils/leaf-map.test.ts:9]: `require` is used here such that `sinon.stub` could work

**Review by matjazv — APPROVED** [review]
Looks good from my side!

**Review by gkoumout — APPROVED** [review]
Looks good to me. I am not so familiar with the topic, but checked all occurrences of synchronous function calls  being replaced. 

Small comment, still in files `packages/tree-builder/src/applications/example/create_accounts.ts` and `packages/tree-builder/src/applications/example/sign_accounts.ts` we are using synchronous functions, but as far as I understand this is intentional, so no need to change.

**Pull request merged**